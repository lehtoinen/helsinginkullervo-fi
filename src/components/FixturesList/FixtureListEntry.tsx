import React from 'react';

import * as styles from './FixtureListEntry.module.css';
import { Fixture } from '../../types';

const FixtureListEntry = ({
  time,
  isCompleted,
  competition,
  homeTeam,
  awayTeam,
  venue,
  city,
  homeScore = 0,
  awayScore = 0,
}: Fixture) => {
  const timeStr = time.substr(0, time.length - 3);
  const winner = isCompleted ? homeScore - awayScore : 0;

  return (
    <div
      className={`${styles.fixture} ${isCompleted ? styles.completed : null}`}
    >
      <div className={styles.subject}>
        <span> {competition}: </span>
        <span className={winner > 0 ? styles.winner : null}>
          {homeTeam} {homeScore}
        </span>{' '}
        -{' '}
        <span className={winner < 0 ? styles.winner : null}>
          {awayScore} {awayTeam}
        </span>
      </div>
      <div className={styles.info}>
        klo {timeStr}{' '}
        <span className={styles.location}>
          {venue}
          {city ? `, ${city}` : ''}
        </span>
      </div>
    </div>
  );
};

export default FixtureListEntry;
