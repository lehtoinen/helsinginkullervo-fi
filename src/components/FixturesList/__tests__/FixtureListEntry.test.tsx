import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import 'jest-axe/extend-expect';

import FixtureListEntry from '../FixtureListEntry';

describe('Fixture', () => {
  const component = ({
    awayScore = 0,
    awayTeam = 'Kullervo/Überkleber',
    competition = 'Regions Cup',
    homeScore = 8,
    homeTeam = 'RiPS',
    isCompleted = true,
    time = '15:00:00',
    venue = 'Riihimäki Keskuskenttä TN',
  }) => (
    <FixtureListEntry
      {...{
        awayScore,
        awayTeam,
        competition,
        homeScore,
        homeTeam,
        isCompleted,
        time,
        venue,
      }}
    />
  );

  it('renders', async () => {
    const { asFragment, container } = render(component({}));
    expect(asFragment()).toMatchSnapshot();

    const axeResults = await axe(container);
    expect(axeResults).toHaveNoViolations();
  });

  it('renders with only required props defined', async () => {
    const { asFragment, container } = render(
      component({
        awayTeam: 'Kullervo/Überkleber',
        competition: 'Regions Cup',
        homeTeam: 'RiPS',
        isCompleted: true,
        time: '15:00:00',
      })
    );
    expect(asFragment()).toMatchSnapshot();

    const axeResults = await axe(container);
    expect(axeResults).toHaveNoViolations();
  });
});
