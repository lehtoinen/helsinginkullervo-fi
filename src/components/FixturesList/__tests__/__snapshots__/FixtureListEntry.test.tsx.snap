// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixture renders 1`] = `
<DocumentFragment>
  <div
    class="undefined undefined"
  >
    <div>
      <span>
         Regions Cup: 
      </span>
      <span>
        RiPS 8
      </span>
       - 
      <span>
        0 Kullervo/Überkleber
      </span>
    </div>
    <div>
      klo 15:00 
      <span>
        Riihimäki Keskuskenttä TN
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Fixture renders with only required props defined 1`] = `
<DocumentFragment>
  <div
    class="undefined undefined"
  >
    <div>
      <span>
         Regions Cup: 
      </span>
      <span>
        RiPS 8
      </span>
       - 
      <span>
        0 Kullervo/Überkleber
      </span>
    </div>
    <div>
      klo 15:00 
      <span>
        Riihimäki Keskuskenttä TN
      </span>
    </div>
  </div>
</DocumentFragment>
`;
